grails-app/services/au/org/ala/biocache/hubs/WebServicesService.groovy

```diff
@@ -507,4 +523,125 @@ class WebServicesService {
         def ois = new ObjectInputStream(bin)
         return ois.readObject()
     }
+
+    JSONElement facetSearch(SearchRequestParams requestParams) {
+        requestParams.pageSize = 0
+        def url = "${grailsApplication.config.biocache.baseUrl}/occurrences/search?${requestParams.getEncodedParams()}"
+        def result = getJsonElements(url)
+
+
+        return result
+    }
+
+    String facetCSVDownload(SearchRequestParams requestParams) {
+        def url = "${grailsApplication.config.biocache.baseUrl}/occurrences/facets/download?${requestParams.getEncodedParams()}&count=true&lookup=true"
+        def result = getText(url)
+        return result
+    }
+
+    def getAllOccurrenceFields() {
+        def url = "${grailsApplication.config.biocache.baseUrl}/index/fields"
+        return getJsonElements(url)?.collect {it.name}
+    }
+
+    @Cacheable('longTermCache')
+    def getMessagesPropertiesFile() {
+        def url = "${grailsApplication.config.biocache.baseUrl}/facets/i18n"
+
+        def map = [:]
+        def lineContent
+        // split text to get lines
+        def lines = getText(url).split("\\r?\\n")
+        lines?.each {
+            // if not comment
+            if (!it.startsWith('#')) {
+                lineContent = it.split('=')
+                if (lineContent.length == 2) {
+                    map[lineContent[0]] = lineContent[1]
+                }
+            }
+        }
+        return map
+    }
+
+    @Cacheable('longTermCache')
+    def getAssertionCodeMap() {
+        JSONArray codes = grailsApplication.mainContext.getBean('webServicesService').getAssertionCodes() // code <-> name
```

Comment:
Why bother pushing out to another method and then having to use application context to call again, when this method only has 2 extra lines (compared to just using all the code in `getAllCodes` here)??? Seems unnecessarily complicated.

Reply:
Unless its for unit testing - can't see any though.

Reply:
Using the `context.getBean()` will be to ensure the `@Cacheable` annotation triggers on `getAssertionCodes` (cache plugin uses Spring Proxies until v4 when they switch to AST transforms).

Also, FWIW webServiceService also used this technique in `init()`

Reply:
Yeah I get that - but this method just calls the other method with an additional 2 lines - maybe I'm missing something but why not just inline that second method inside `getAssertionCodeMap` - it will only get called when cache is purged either way.

Reply:
Good point - I thought `getAssertionCodes` was called independently but it's only used by this method so inlining it makes sense.

Reply:
it's inlined.